ifupdown2/ifupdown/networkinterfaces.py

```diff
@@ -204,21 +206,31 @@ def process_auto(self, lines, cur_idx, lineno):
             self._parse_error(self._currentfile, lineno,
                     'invalid auto line \'%s\''%lines[cur_idx])
             return 0
+
+        if "auto" not in self.allow_classes:
+            self.allow_classes["auto"] = []
+
         for a in auto_ifaces:
             if a == 'all':
                 self.auto_all = True
+                self.allow_classes["allow"].extend(auto_ifaces)
                 break
             r = utils.parse_iface_range(a)
             if r:
                 if len(r) == 3:
                     # eg swp1.[2-4], r = "swp1.", 2, 4)
                     for i in range(r[1], r[2]+1):
-                        self.auto_ifaces.append('%s%d' %(r[0], i))
+                        ifname = '%s%d' %(r[0], i)
+                        self.auto_ifaces.append(ifname)
+                        self.allow_classes["allow"].append(ifname)
                 elif len(r) == 4:
                     for i in range(r[1], r[2]+1):
                         # eg swp[2-4].100, r = ("swp", 2, 4, ".100")
-                        self.auto_ifaces.append('%s%d%s' %(r[0], i, r[3]))
+                        ifname = '%s%d%s' %(r[0], i, r[3])
+                        self.auto_ifaces.append(ifname)
+                        self.allow_classes["auto"].append(ifname)
             self.auto_ifaces.append(a)
+            self.allow_classes["auto"].append(a)
```

Comment:
Also :wink: I would like to suggest changing the existing code here in order to enhance its readability. Take a look at the code snippet below:

```py
            r = utils.parse_iface_range(a)
            if r:
                prefix, start, end = r[0], r[1], r[2] + 1
                suffix = '' if len(r) <= 3 else r[3]
                ifnames = [f'{prefix}{i}{suffix}' for i in range(start, end)]
                self.auto_ifaces.extend(ifnames)
                self.allow_classes["allow"].extend(ifnames)
            self.auto_ifaces.append(a)
            self.allow_classes["auto"].append(a)

```


Hope to see this PR being merged. :-)

Reply:
Please note that this is just an example and it only works for the first chunk of `allow_classes['allow']`. It won't function as it did before. I don't fully understand why we should extend the "auto" class when there is a suffix, and the "allow" class when there isn't. However, I haven't delved deeply into the code yet.